dataplug/app/com/hubofallthings/dataplug/controllers/Application.scala

```diff
@@ -175,9 +179,21 @@ class Application @Inject() (
       .recover {
         case e =>
           logger.error(s"$provider API cannot be accessed: ${e.getMessage}. Redirecting to $provider OAuth service.", e)
-          Redirect(com.hubofallthings.dataplug.controllers.routes.SocialAuthController.authenticate(provider))
+          Redirect(dataPlugViewSet.disconnectRedirect)
       }
   }
 
+  def disconnect(): Action[AnyContent] = SecuredAction.async { implicit request =>
+    val userId = request.identity.linkedUsers.head.userId
+    val phata = request.identity.userId
+    for {
+      redirect <- deactivateEndpoints()(request)
+      _ <- userService.delete(phata, userId)
```

Comment:
I have not looked, but since `deletes` are a big thing, are we logging the actions in each of these steps?

Reply:
That's a very good point and it skipped my mind. It's a good tactic to use logs of such actions for future reference. I think that's something I need to add as well. 👍

Reply:
👍

Reply:
@whiteshadow-gr not fixed yet?

Reply:
Still no logging in the disconnect handler?

Reply:
We do log it. look at UserDaoImp file. Or my next comment